File: tests/fixtures/sample.ts
import { helper } from './helper';

interface Runner {
  run(): number;
}

class Parser implements Runner {
  parse(text: string): number {
    return text.length;
  }
  run(): number {
    return this.parse('hi');
  }
}

type Id = string;

function run(p: Parser): number {
  return p.parse('hi');
}

File: tests/fixtures/sample.rs
use std::collections::HashMap;

pub struct Parser {
    pub size: usize,
}

impl Parser {
    pub fn parse(&self, text: &str) -> usize {
        text.len()
    }
}

pub fn run(p: &Parser) -> usize {
    p.parse("hi")
}
